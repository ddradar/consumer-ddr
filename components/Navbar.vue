<template>
  <nav class="navbar is-primary" role="navigation" aria-label="main navigation">
    <div class="navbar-brand">
      <nuxt-link class="navbar-item has-text-weight-bold" to="/">
        <img class="images" src="~assets/img/logo.svg" alt="Logo" />
        Consumer DDR
      </nuxt-link>

      <a
        role="button"
        class="navbar-burger"
        aria-label="menu"
        aria-expanded="false"
        data-target="navbarMenu"
      >
        <span aria-hidden="true"></span>
        <span aria-hidden="true"></span>
        <span aria-hidden="true"></span>
      </a>
    </div>

    <div id="navbarMenu" class="navbar-menu">
      <div class="navbar-start">
        <nuxt-link class="navbar-item" to="/song/">Songs</nuxt-link>
      </div>
      <div class="navbar-end" />
    </div>
  </nav>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

export interface MenuItem {
  title: string
  subMenu: { name: string; href: string }[]
}

@Component
export default class Navbar extends Vue {
  @Prop({ required: true }) readonly menuList!: MenuItem[]
}
</script>
