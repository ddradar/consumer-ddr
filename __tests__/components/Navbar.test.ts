import { mount, RouterLinkStub } from '@vue/test-utils'
import { describe, expect, test } from 'vitest'

import Navbar from '~~/components/Navbar.vue'

import { plugins } from '../test-utils'

describe('components/Navbar.vue', () => {
  const global = {
    plugins,
    stubs: { NuxtLink: RouterLinkStub, SearchBox: true },
  }

  test('renders correctly', () => {
    // Arrange - Act
    const wrapper = mount(Navbar, { global })

    // Assert
    expect(wrapper.element).toMatchSnapshot()
  })

  describe('navbar-burger', () => {
    test('@click changes "is-active" class', async () => {
      // Arrange
      const wrapper = mount(Navbar, { global })

      // Act
      const burger = wrapper.find('.navbar-burger')
      const menu = wrapper.find('#navbarMenu')
      await burger.trigger('click')

      // Assert
      expect(burger.classes()).toContain('is-active')
      expect(menu.classes()).toContain('is-active')
    })
    test('@click: 2 reverts "is-active" class', async () => {
      // Arrange
      const wrapper = mount(Navbar, { global })

      // Act
      const burger = wrapper.find('.navbar-burger')
      const menu = wrapper.find('#navbarMenu')
      await burger.trigger('click')
      await burger.trigger('click')

      // Assert
      expect(burger.classes()).not.toContain('is-active')
      expect(menu.classes()).not.toContain('is-active')
    })
  })
})
