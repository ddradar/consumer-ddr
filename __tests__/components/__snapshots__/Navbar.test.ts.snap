// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/Navbar.vue > renders correctly 1`] = `
<nav
  aria-label="main navigation"
  class="navbar is-primary"
  role="navigation"
>
  <div
    class="navbar-brand"
  >
    <a
      class="navbar-item has-text-weight-bold"
    >
      <img
        alt="Logo"
        class="images"
        src="/assets/img/logo.svg"
      />
      <span
        class="is-hidden-mobile"
      >
        Consumer DDR
      </span>
    </a>
    <search-box-stub
      class="navbar-item"
    />
    <a
      aria-expanded="false"
      aria-label="menu"
      class="navbar-burger"
      data-target="navbarMenu"
      role="button"
    >
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
    </a>
  </div>
  <div
    class="navbar-menu"
    id="navbarMenu"
  >
    <div
      class="navbar-start"
    >
      <a
        class="navbar-item"
      >
        Songs
      </a>
    </div>
  </div>
</nav>
`;
